pyomo/contrib/mindtpy/nlp_solve.py

```diff
@@ -139,7 +138,7 @@ def handle_nlp_subproblem_tc(fixed_nlp, result, solve_data, config, cb_opt=None)
         solve_data.should_terminate = True
     else:
         handle_subproblem_other_termination(fixed_nlp, result.solver.termination_condition,
-                                            solve_data, config)
+                                            solve_data, config, cb_opt)
 
 
 # The next few functions deal with handling the solution we get from the above NLP solver function
```

Comment:
I'm a bit confused about your design here. Several (if not all?) of the functions below have counterparts with very similar code in the algorithm base class (like `handle_subproblem_optimal`). When are you using these implementations and when are you using the ones on the class, and can they be combined?

Reply:
Nope. The functions here are used by the single-tree implementation, which can not call the methods in the algorithm base class.

Reply:
Ah, okay. Let's keep this in mind for a future single-tree design discussion then.